drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -101,6 +113,30 @@ static int adc_smartbond_channel_setup(const struct device *dev,
 	return 0;
 }
 
+static inline void gpadc_smartbond_pm_policy_state_lock_get(struct adc_smartbond_data *data)
+{
```

Comment:
Based on a discussion with @ydamigos, the device runtime mode is not handled correctly. According to the [Zephyr guidelines](https://docs.zephyrproject.org/latest/services/pm/device_runtime.html), a device should explicitly call ` pm_device_runtime_get` before a conversion takes place and  `pm_device_runtime_put` right after the conversions are fully exercised. I think these two APIs could be encapsulated in `gpadc_smartbond_pm_policy_state_lock_get` and `gpadc_smartbond_pm_policy_state_lock_put` respectively safeguarded by `CONFIG_PM_DEVICE_RUNTIME`.

Reply:
This is implemented now as described